capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java

```diff
@@ -1,166 +1,134 @@
-import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
 
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that allows users to post and get clubs. */
+
 @WebServlet("/api/clubs")
 public class ClubServlet extends HttpServlet {
-
-  private JsonObject createRequestBodyJson(HttpServletRequest request) {
-    JsonObject jsonObject = new JsonObject();
-    StringBuffer jb = new StringBuffer();
-    String line = null;
-    try {
-      BufferedReader reader = new BufferedReader(new InputStreamReader((request.getInputStream())));
-      while ((line = reader.readLine()) != null) {
-       jb.append(line);
-      }
-    } catch (Exception e) { 
-      System.err.println("Error: " + e);
-    }
-    try {
-      Gson gson = new Gson();
-      JsonElement data = JsonParser.parseString(jb.toString());
-      jsonObject = data.getAsJsonObject();
-    } catch (Exception e) {
-      System.err.println("Error parsing JSON request string");
-    }
-
-    return jsonObject;
+  private Firestore db; 
+  private CollectionReference clubs;
+  private Gson gson;
+
+  public ClubServlet() throws IOException {
+    db = Utility.getFirestoreDb();
+    clubs = db.collection("clubs");
+    gson = new Gson();
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Club");
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
-    List<Entity> entities = results.asList(FetchOptions.Builder.withLimit(5));
-
-    List<Club> clubs = new ArrayList<>();
-    for(Entity e : entities) {
-      long id = (long) e.getKey().getId();
-      String name = (String) e.getProperty("name");
-      String description = (String) e.getProperty("description");
-      String announcement = (String) e.getProperty("announcement");
-
-      @SuppressWarnings("unchecked") // Cast can't verify generic type.
-      Collection<String> posts = (ArrayList<String>) e.getProperty("posts");
-
-      @SuppressWarnings("unchecked") // Cast can't verify generic type.
-      Collection<Long> memberIDs = (HashSet<Long>) e.getProperty("memberIDs");
-      
-      @SuppressWarnings("unchecked") // Cast can't verify generic type.
-      Collection<Long> inviteIDs = (HashSet<Long>) e.getProperty("inviteIDs");
-
-      long ownerID = (long) e.getProperty("ownerID");
-      String gbookID = (String) e.getProperty("gbookID");
-      Club club = new Club(id, name, description, announcement, ownerID, gbookID);
-      clubs.add(club);
+    List<Club> result = new ArrayList<>();
+    if (request.getParameter("id") != null) {
+      DocumentReference docRef = clubs.document(request.getParameter("id"));
+      ApiFuture<DocumentSnapshot> future = docRef.get();
```

Comment:
I wasn't sure what to name it since an ApiFuture represents the result of an asynchronous computation (which is resolved when I call future.get()), so it isn't really anything but an intermediary. I'll rename it something like ``` asyncDocument ``` so that it is more informative for the reader.